### PR TITLE
Add protocol adapter overview

### PR DESCRIPTION
**Pull request by tbedford**
## Description

Adds a protocol adapter overview topic at `/protocol-adapters`.

* [EDX-122](https://ably.atlassian.net/browse/EDX-122).

## Review

If reviewing locally the overview can be found at:

http://localhost:4000/protocol-adapters/

### CODE REVIEW

File: content/protocol-adapters/index.textile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Each of these will be described in the following sections.
+
+h2(#mqtt). MQTT
+
+MQTT protocol adapter provides an interface between an MQTT client and Ably. MQTT clients can subscribe and publish to Ably channels. Any client that supports MQTT can therefore leverage Ably's realtime network, with some constraints. 
+
+For more information on Ably's MQTT protocol adapter see the "MQTT documentation":/mqtt.
+
+h2(#sse). SSE
+
```
Comment (Srushtika) [inline content/protocol-adapters/index.textile:27]: It's worth noting that SSE is subscribe-only, and that is one of the considerations based on which one can make a decision to either go with ably-js or SSE for example.

File: content/protocol-adapters/index.textile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+While the recommended method for connecting to Ably is through the use of the Ably Client Library SDKs, there are certain situation where this is not possible, due to lack of platform support, or platform resource constraints. In this situation, certain clients may be supported though the use of Ably protocol adapters. There are two protocol adapters in this section of the documentation:
+
+1. MQTT
+2. SSE
+
+Each of these will be described in the following sections.
+
+h2(#mqtt). MQTT
+
+MQTT protocol adapter provides an interface between an MQTT client and Ably. MQTT clients can subscribe and publish to Ably channels. Any client that supports MQTT can therefore leverage Ably's realtime network, with some constraints.
```
Comment (Srushtika) [inline content/protocol-adapters/index.textile:22]: Worth mentioning 'IoT' as a use case due to battery and bandwidth constraints but we can skip an in-depth explanation.

File: content/protocol-adapters/index.textile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+index: 0
+jump_to:
+  Help with:
+    - MQTT#mqtt
+    - SSE#sse
+---
+
+While the recommended method for connecting to Ably is through the use of the Ably Client Library SDKs, there are certain situations where this is not possible, due to lack of platform support, or platform resource constraints. In this case, certain clients may be supported through the use of Ably protocol adapters. 
+
+Ably Client Library SDKs are the recommended method for connecting to Ably, because they offer support for a comprehensive set of Ably features, such as automatic connection management, authentication token renewal via @authUrl@, and presence, whereas the protocol adapters may not support these features: for example MQTT protocol adapter does not have support for presence, and the SSE protocol adapter does not have support for automatic token renewal (this has to be handled manually in the client code). A full list of Ably Client Library SDKs can be found on the "SDK page":https://ably.com/download, and features supported by each SDK can be found in the "features matrix":https://ably.com/download/sdk-feature-support-matrix.
```
Comment (Srushtika) [inline content/protocol-adapters/index.textile:15]: Would be good to link to each of these features, i.e. connection management, token, presence, etc

File: content/protocol-adapters/index.textile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+1. MQTT
+2. SSE
+
+Each of these will be described briefly in the following sections.
+
+h2(#mqtt). MQTT
+
+"MQTT":https://mqtt.org/ is an OASIS standard messaging protocol for the Internet of Things (IoT). It is designed as an extremely lightweight publish/subscribe messaging transport that is ideal for connecting remote devices with a small code footprint and minimal network bandwidth.
+
+The Ably MQTT protocol adapter provides an interface between an MQTT client and Ably. Being a protocol designed for small systems and IoT devices, MQTT has a low overhead in terms of system resource requirements. MQTT libraries are widely available, and support a range of languages such as C/C++, Python, and Rust. Because of this, client platforms and programming languages where an Ably Client Library SDK is not available, can still access the Ably realtime network through the MQTT client and the Ably MQTT protocol adapter.
```
Comment (Srushtika) [inline content/protocol-adapters/index.textile:30]: Should it be Ably Realtime? (capital R)

File: content/protocol-adapters/index.textile [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+MQTT clients can subscribe and publish to Ably channels. Any client that supports MQTT can therefore leverage Ably's realtime network, with some constraints. 
+
+For more information on Ably's MQTT protocol adapter see the "MQTT documentation":/mqtt.
+
+h2(#sse). SSE
+
+Server-Sent Events (SSE) is a functionality built into more recent versions of browsers. Because "SSE support":https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events is built-in, it does not require the loading of any additional libraries, such as is the case if you were to use the Ably JavaScript Client Library SDK. You will have to handle certain operations in your client, for example, token renewal will need to be managed by your client code.
```
Comment (Srushtika) [inline content/protocol-adapters/index.textile:38]: Worth explaining what you mean by 'more recent versions of browsers'

Comment (Srushtika) [reply content/protocol-adapters/index.textile:38]: Would be good to link to Ably JS client lib sdk

File: content/protocol-adapters/index.textile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+The Ably SSE protocol adapter allows the browser to subscribe to a realtime stream of events from an Ably channel. Publishing to a channel is not supported by SSE.
+
+In addition to the SSE endpoint, Ably also provides a raw HTTP streaming endpoint.
```
Comment (Srushtika) [inline content/protocol-adapters/index.textile:42]: Worth explaining in a single line what the difference is between the two, the readers will have no context at this point.

File: content/protocol-adapters/index.textile [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+meta_keywords: "SSE, MQTT, IoT, thin client, protocol adapter, lightweight clients, Arduino, Raspberry Pi, wearable"
+section: protocol-adapters
+index: 0
+jump_to:
+  Help with:
+    - MQTT#mqtt
+    - SSE#sse
+---
+
+While the recommended method for connecting to Ably is through the use of the Ably Client Library SDKs, there are certain situations where this is not possible, due to lack of platform support, or platform resource constraints. In this case, certain clients may be supported through the use of Ably protocol adapters.
```
Comment (Srushtika) [inline content/protocol-adapters/index.textile:13]: It seems a bit repetitive to have this para and the next explaining almost the same thing, wdyt?

**Review by Srushtika — CHANGES_REQUESTED** [review]
@m-hulbert think one for you to finish up - I've added a few comments, please let me know if you have any questions

File: content/protocol-adapters/index.textile (hunk 3 above)
Comment (m-hulbert) [reply content/protocol-adapters/index.textile]: Added links for each item.

File: content/protocol-adapters/index.textile [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+There are two protocol adapters covered in this section of the documentation:
+
+1. MQTT
+2. SSE
+
+Each of these will be described briefly in the following sections.
+
+h2(#mqtt). MQTT
+
+"MQTT":https://mqtt.org/ is an OASIS standard messaging protocol for the Internet of Things (IoT). It is designed as an extremely lightweight publish/subscribe messaging transport that is ideal for connecting remote devices with a small code footprint and minimal network bandwidth.
```
[the remark this answers is not in the document]
Comment (m-hulbert) [reply content/protocol-adapters/index.textile]: Added a link to OASIS.

File: content/protocol-adapters/index.textile (hunk 5 above)
Comment (m-hulbert) [reply content/protocol-adapters/index.textile]: Reworded to "modern browsers" which I think is sufficient given the link in the next sentence to its support. I've also linked out the JS GitHub repo.

File: content/protocol-adapters/index.textile [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+MQTT clients can subscribe and publish to Ably channels. Any client that supports MQTT can therefore leverage Ably's realtime network, with some constraints. 
+
+For more information on Ably's MQTT protocol adapter see the "MQTT documentation":/mqtt.
+
+h2(#sse). SSE
+
+Server-Sent Events (SSE) is a functionality built into more recent versions of browsers. Because "SSE support":https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events is built-in, it does not require the loading of any additional libraries, such as is the case if you were to use the Ably JavaScript Client Library SDK. You will have to handle certain operations in your client, for example, token renewal will need to be managed by your client code.
+
+The Ably SSE protocol adapter allows the browser to subscribe to a realtime stream of events from an Ably channel. Publishing to a channel is not supported by SSE.
```
[the remark this answers is not in the document]
Comment (m-hulbert) [reply content/protocol-adapters/index.textile]: Re-ordered this section so it reads more clearly about info vs drawbacks.

File: content/protocol-adapters/index.textile (hunk 7 above)
Comment (m-hulbert) [reply content/protocol-adapters/index.textile]: I've re-written the introductory section to account for this.

File: content/protocol-adapters/index.textile (hunk 6 above)
Comment (m-hulbert) [reply content/protocol-adapters/index.textile]: Done!

**Review by Srushtika — APPROVED** [review]
Looks good to me. Thanks, @m-hulbert and @tbedford. Can we merge this in?

**Pull request merged**